cmd/bucket-replication-stats.go

```diff
@@ -0,0 +1,143 @@
+/*
+ * MinIO Cloud Storage, (C) 2021 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+
+	"github.com/minio/minio/pkg/bucket/replication"
+)
+
+// BucketReplicationStats represents inline replication statistics
+// such as pending, failed and completed bytes in total for a bucket
+type BucketReplicationStats struct {
+	// Pending size in bytes
+	PendingSize uint64 `json:"objectsPendingReplicationTotalSize"`
+	// Completed size in bytes
+	ReplicatedSize uint64 `json:"objectsReplicatedTotalSize"`
+	// Total Replica size in bytes
+	ReplicaSize uint64 `json:"objectsReplicaTotalSize"`
+	// Failed size in bytes
+	FailedSize uint64 `json:"objectsFailedReplicationTotalSize"`
+	// Total number of pending operations including metadata updates
+	OperationsPendingCount uint64 `json:"objectsPendingReplicationCount"`
+}
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-metrics.html we should have at least two of these published by AWS S3 part of this. 

We should add how many objects didn't replicate in time? such as objects that missed SLA